internal/cli/delete.go

```diff
@@ -18,14 +25,39 @@ func NewCmdDelete() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			return RunDelete(kind, name)
+
+			if cmd.Flags().Lookup("owner").Changed && kind != TemplateVersionKind {
```

Comment:
feels like all this logic should be part of a verify method. In general, cobra exposes flags that populate some object then we verify values against defaults. This can be a followup.

Reply:
Do you mean to use PreRunE for validations rather than RunE?